src/specklepy/objects/geometry/mesh.py

```diff
@@ -0,0 +1,210 @@
+from dataclasses import dataclass, field
+from typing import List, Tuple
+
+from specklepy.objects.base import Base
+from specklepy.objects.geometry.point import Point
+from specklepy.objects.interfaces import IHasArea, IHasUnits, IHasVolume
+
+
+@dataclass(kw_only=True)
+class Mesh(
+    Base,
+    IHasArea,
+    IHasVolume,
+    IHasUnits,
+    speckle_type="Objects.Geometry.Mesh",
+    detachable={"vertices", "faces", "colors", "textureCoordinates"},
+    chunkable={
+        "vertices": 31250,
+        "faces": 62500,
+        "colors": 62500,
+        "textureCoordinates": 31250,
+    },
+):
+    """
+    a 3D mesh consisting of vertices and faces with optional colors and texture coordinates
+    """
+    vertices: List[float]
+    faces: List[int]
+    colors: List[int] = field(default_factory=list)
+    textureCoordinates: List[float] = field(default_factory=list)
+
+    def __repr__(self) -> str:
+        return (
+            f"{self.__class__.__name__}("
+            f"vertices: {self.vertices_count}, "
+            f"faces: {self.faces_count}, "
+            f"units: {self.units}, "
+            f"has_colors: {len(self.colors) > 0}, "
+            f"has_texture_coords: {len(self.textureCoordinates) > 0})"
+        )
+
+    @property
+    def vertices_count(self) -> int:
+        """
+        get the number of vertices in the mesh
+        """
+        if not self.vertices:
+            return 0
+        if len(self.vertices) % 3 != 0:
+            raise ValueError(
+                f"Invalid vertices list: length ({len(
+                    self.vertices)}) must be a multiple of 3"
+            )
+        return len(self.vertices) // 3
+
+    @property
+    def faces_count(self) -> int:
```

Comment:
I don't think we should have this property. 
Its not in the C#, and is not performant to have to loop through the list to calculate.